### PR TITLE
Add TempData property and some functional tests

### PR DESCRIPTION
**Pull request by rynowak**
/cc @pranavkm @ryanbrandenburg

### CODE REVIEW

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesTest.cs [hunk 1]
```diff
@@ -80,5 +82,68 @@ public RazorPagesTest(MvcTestFixture<RazorPagesWebSite.Startup> fixture)
...
+            var request = new HttpRequestMessage(HttpMethod.Get, url);
+            
+            // Act 1
+            var response = await Client.SendAsync(request);
+            
+            // Assert 1
+            Assert.Equal(HttpStatusCode.Redirect, response.StatusCode);
+
+            // Arrange 2
+            url = new Uri(new Uri("http://localhost"), response.Headers.Location).AbsoluteUri;
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesTest.cs]: We set the BaseAddress on the Client. You might not need this concatenation.

File: src/Microsoft.AspNetCore.Mvc.RazorPages/Page.cs [hunk 2]
```diff
@@ -90,5 +93,14 @@ public override void EndContext()
             }
         }
 
+        protected IActionResult Redirect(string url)
+        {
+            return new RedirectResult(url);
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.RazorPages/Page.cs]: Null or empty check - https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/ControllerBase.cs#L362-L365

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesTest.cs (hunk 1 above)
Comment (rynowak) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesTest.cs]: Ok dope, I'll try without this.

**Comment by rynowak** [comment]
I created a new item to track porting over other `IActionResult` functionality. I'm adding docs for the things that I added.

**Pull request merged**